R/graph_methods.R

```diff
@@ -62,6 +62,28 @@ topological_sort <- function(graph) {
   }
 }
 
+# get the descendants (all children) of node, given list mapping parent to children
+# and their distances
+get_descendants_distance <- function(parents_to_children, node) {
+  internal_recursive_fun <- function(parents_to_children, node, distance = 0L) {
+    if (is.null(parents_to_children[[node]])) {
+      return()
+    }
+    c(
+      lapply(parents_to_children[[node]], list, distance + 1L),
+      unlist(lapply(parents_to_children[[node]],
+                    internal_recursive_fun, parents_to_children = parents_to_children,
+                    distance = distance + 1L), recursive = FALSE)
+    )
+  }
+  internal_recursive_fun(parents_to_children, node) %>%
```

Comment:
this is a bad name for the function
then, you can just call it `f`, that is equally informative (and maybe make a comment that it is recursively called)

Reply:
blame @pawelru  :)

Reply:
Changed to an iterative not recursive function